src/baldr/nodeinfo.cc

```diff
@@ -84,14 +84,21 @@ NodeInfo::NodeInfo(const PointLL& tile_corner,
 }
 
 // Sets the latitude and longitude.
-void NodeInfo::set_latlng(const PointLL& tile_corner, const std::pair<float, float>& ll) {
+void NodeInfo::set_latlng(const PointLL& tile_corner, const PointLL& ll) {
   // Protect against a node being slightly outside the tile (due to float roundoff)
-  lat_offset_ = ll.second < tile_corner.lat()
-                    ? 0
-                    : static_cast<uint32_t>((ll.second - tile_corner.lat()) / kDegreesPrecision);
-  lon_offset_ = ll.first < tile_corner.lng()
-                    ? 0
-                    : static_cast<uint32_t>((ll.first - tile_corner.lng()) / kDegreesPrecision);
+  lat_offset_ = 0;
+  if (ll.lat() > tile_corner.lat()) {
+    auto lat = std::round((ll.lat() - tile_corner.lat()) / 1e-7);
+    lat_offset_ = lat / 10;
```

Comment:
old code will still work because the first 22 bits get the 6 digit lat/lon as they did before then below we also set the 4 extra bits with the 7th digit